module/VuFind/src/VuFind/Section/SectionService.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+/**
+ * Section service.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Section
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Section;
+
+use VuFind\Config\Feature\ConfigSettingPropertiesInterface;
+use VuFind\Config\YamlReader;
+use VuFind\Exception\BadConfig;
+use VuFind\Exception\ConfigException;
+use VuFind\Navigation\PluginManager as NavigationPluginManager;
+use VuFind\Section\PluginManager as SectionPluginManager;
+
+use function is_string;
+
+/**
+ * Section service.
+ *
+ * @category VuFind
+ * @package  Section
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class SectionService implements SectionServiceInterface
+{
+    /**
+     * Constructor.
+     *
+     * @param YamlReader              $yamlReader      YAML reader
+     * @param SectionPluginManager    $sectionPm       Section plugin manager
+     * @param NavigationPluginManager $navigationPm    Navigation plugin manager
+     * @param string                  $userLocale      User locale
+     * @param array                   $fallbackLocales Fallback locales
+     */
+    public function __construct(
+        protected YamlReader $yamlReader,
+        protected SectionPluginManager $sectionPm,
+        protected NavigationPluginManager $navigationPm,
```

Comment:
I'd suggest that, while slightly longer, maybe $sectionManager and $navigationManager would be more readable property names.